File: 02_ProgrammingFundamentals/_10_TextProcessing/_02_Exercise/_01_ValidUsernames.java
package _02_ProgrammingFundamentals._10_TextProcessing._02_Exercise;

import java.util.Scanner;

public class _01_ValidUsernames {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        // Read input values
        String[] console = scan.nextLine().split(", ");

        // Validate usernames
        for (int i = 0; i < console.length; i++) {
            // Validate the length
            if (console[i].length() >= 3 && console[i].length() <= 16) {
                boolean isValid = true;

                // Validate characters
                for (int j = 0; j < console[i].length(); j++) {
                    char symbol = console[i].charAt(j);

                    isValid = Character.isLetterOrDigit(symbol) || symbol == '-' || symbol == '_';
                    if (!isValid) {
                        break;
                    }
                }

                // Print result
                if (isValid) {
                    System.out.println(console[i]);
                }
            }
        }

        scan.close();
    }
}

File: 24_C#Fundamentals/_08_TextProcessing/_02_Exercise/_01_ValidUsernames/_01_ValidUsernames.cs
using System;

public class _01_ValidUsernames
{
    public static void Main(string[] args)
    {
        string[] console = Console.ReadLine().Split(", ");

        // Validate usernames
        for (int i = 0; i < console.Length; i++)
        {
            // Validate the length
            if (console[i].Length >= 3 && console[i].Length <= 16)
            {
                bool isValid = true;

                // Validate characters
                for (int j = 0; j < console[i].Length; j++)
                {
                    char symbol = console[i][j];

                    isValid = char.IsLetterOrDigit(symbol) || symbol == '-' || symbol == '_';
                    if (!isValid)
                    {
                        break;
                    }
                }

                // Print result
                if (isValid)
                {
                    Console.WriteLine(console[i]);
                }
            }
        }
    }
}
